bin/ota-validate.js

```diff
@@ -9,49 +8,71 @@ import Mocha from 'mocha';
 
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
-const VALIDATE_TEST_FILEPATH = '../scripts/declarations/validate/index.mocha.js';
-const VALIDATE_PATH = path.resolve(__dirname, VALIDATE_TEST_FILEPATH);
+export function createMocha({ delay = false, reporter = 'spec' } = {}) {
+  return new Mocha({
+    delay,
+    failZero: true,
+    reporter,
+  });
+}
+
+export async function runMochaTests(mocha, testPath) {
+  try {
+    mocha.addFile(testPath); // With `delay` option, this statement will not load the file directly, `loadFilesAsync` is required.
+    await mocha.loadFilesAsync(); // Load files previously added to the Mocha cache with `addFile`.
+
+    return new Promise(resolve => {
+      let hasFailedTests = false;
+
+      mocha.run()
+        .on('fail', () => { hasFailedTests = true; })
+        .on('end', () => { resolve(hasFailedTests ? 1 : 0); });
+    });
+  } catch (error) {
+    console.error('Error running tests:', error);
+
+    return 2;
+  }
+}
 
-// Mocha catches unhandled rejection from the user code and re-emits them to the process (see https://github.com/mochajs/mocha/blob/master/lib/runner.js#L198)
-process.on('unhandledRejection', reason => {
-  // Re-throw them so that the validation command fails in these cases (for example, if there is a syntax error when parsing JSON declaration files)
-  throw reason;
+process.on('unhandledRejection', reason => { // Mocha catches unhandled rejection from the user code and re-emits them to the process (see https://github.com/mochajs/mocha/blob/master/lib/runner.js#L198)
```

Comment:
The line number `#L198` referred to here does not agree with the version of Mocha used and there is no guarantee that this link will work over time.